pkgs/development/tools/build-managers/build2/default.nix

```diff
@@ -67,6 +68,10 @@ stdenv.mkDerivation rec {
     "config.bin.lib=${configSharedStatic enableShared enableStatic}"
     "config.cc.poptions+=-I${lib.getDev libpkgconf}/include/pkgconf"
     "config.build2.libpkgconf=true"
+  ] ++ lib.optionals stdenv.isDarwin [
+    # The default "ar" from darwin.cctools-llvm produces an archive on which
+    # the linker chokes. I'm not sure why.
+    "config.bin.ar=${darwin.cctools-port}/bin/ar"
```

Comment:
If we go this route, the setup hook also needs to set `config.bin.ar` to the cctools `ar`. Otherwise, users of build2 will encounter the same linker errors.

Reply:
That makes sense. I'll prepare something here in case we want to go this route.